Homework 20

KhrysCS opened the pull request:
Homework #20: Selenium Waits

Modified BaseTest.java - Replaced timeouts with explicit waits
Added Homeword20.java - Replaced timeouts with explicit waits
Modified TestNG.xml - Added Homeword20 class

KhrysCS committed:
Update LoginTests.java

Homework #15: Java Git

Changed the URL from https://qa.koel.app/" to https://testpro.io/".

KhrysCS committed:
Homework #16: Intro to Selenium WebDriver

Homework #16: Intro to Selenium WebDriver

Created a test case for registrationNavigation.

Added new class named Homework16 which contains the test case for registrationNavigation.

Modified TestNG.xml: added Homework16 class

KhrysCS committed:
Merge pull request #1 from KhrysCS/homework-16

Homework #16: Intro to Selenium WebDriver

KhrysCS committed:
Lesson17 Demos and Homework #17: TestNG Overview

Added ProfileTests.java: Demo class from lesson 17

Added Homework17.java: added method to test adding song to playlist. Completed all functionality. Still need to refactorize.

Modified BaseTest.java: added helper methods
Modified LoginTests.java: refactored class to use helper methods from BaseTest

Modified TestNG.xml: added Homework17 and ProfileTests

KhrysCS committed:
Homework #17: TestNG Overview Completed

Refactorized Homewor17.java and added comments.

KhrysCS committed:
Merge pull request #2 from KhrysCS/homework-17

Homework #17: TestNG Overview

KhrysCS committed:
A51-L18-Class-Demo

Some practice code from lesson 18 and files to keep notes and .

KhrysCS committed:
Homework #18: XML File Configurations

Homework #18: XML File Configurations

Added Test class Homework18 .java the test class contains the playSong test case
Added Test suit  SmokeHW18.xml file for smoke testing the new test class by method.
Modified the build.gradle file to invoke the SmokeHW18.xml file for testing

KhrysCS committed:
Merge pull request #3 from KhrysCS/homework-18

Homework 18

KhrysCS committed:
Update README file

KhrysCS committed:
Homework #19: Additional Annotations & DataProviders

Modified build.gradle to use TestNG.xml
Modifed BaseTest.java to add @Parameter BaseURL and assign parameter to url string.
Modified TestNG config file to add parameter BaseURL and added class name Homework19
Added Homework19.java test class for test case deletePlaylist

KhrysCS committed:
Update ReadMe file

KhrysCS committed:
Merge pull request #4 from KhrysCS/homework-19

Homework 19

KhrysCS committed:
Merge pull request #5 from KhrysCS/homework-15

Update LoginTests.java

KhrysCS committed:
Update LoginTests.java

KhrysCS committed:
Homework #20: Selenium Waits

Homework #20: Selenium Waits

Modified BaseTest.java - Replaced timeouts with explicit waits
Added Homeword20.java - Replaced timeouts with explicit waits
Modified TestNG.xml - Added Homeword20 class

KhrysCS committed:
Create Homework20Draft.java

KhrysCS committed:
Revert "Create Homework20Draft.java"

This reverts commit c7bae0d28194468b545da835d2acc0c4b9a0c32b.

Berichicko commented:
You got this problem, please correct it. Please finish your homework.

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/08fc420e-116e-4c0b-92a8-9d2fffbbe952)

HINT

![Test status_en](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/043743a4-5a5f-494c-bb74-6a7fab53afc1)

![Test failed_en](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/24e65eb1-393f-467e-84b1-35ff0595f187)

KhrysCS committed:
Update Homework20.java

Edited Homework20.java class to add a song to the playlist and then delete the song added,

Berichicko commented on src/test/java/Homework20.java:
```diff
@@ -0,0 +1,71 @@
+
+public class Homework20 extends BaseTest {
+
+    @Test
+    public void deletePlaylist() throws InterruptedException{
+        String uniquePlaylist = "1-HW19-Playlist";
+        String deletePlaylistMsg = "Deleted playlist \"1-HW19-Playlist.\"";
+
+        //Pre-condition
+        //Login
+        provideEmail("demo@class.com");
+        providePassword("te$t$tudent");
+        clickSubmit();
+
+        //Create empty playlist
+        createPlaylist(uniquePlaylist);
+
+        Thread.sleep(4000);
+
+        //Click on empty playlist created
+        clickOnPlaylist();
+
+        //delete selected playlist
+        clickOnDeleteBtn();
+
+        Thread.sleep(2000);
+
+        //Expected Results
+        WebElement notificationMsg = wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("div.alertify-logs div.success")));//driver.findElement(By.cssSelector("div.alertify-logs div.success"));
+
+        Assert.assertEquals(deletePlaylistMsg,notificationMsg.getText());
```
Please, delete `throws InterruptedException`

**Thread. sleep is bad!** It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

KhrysCS committed:
Update Homework20.java

Modified Homework 20 to remove Thread.sleep

Berichicko commented:
Hello,

I have checked your homework 20, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![804deb64dc406a20a75da05d91a1a69e](https://user-images.githubusercontent.com/104761482/235961479-9afb2841-b657-46c9-99f8-2402f933b843.jpg)

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/a9c1b0da-d928-4117-ae48-7a6a217c5328)
